google-cloud-firestore/lib/google/cloud/firestore/query.rb

```diff
@@ -0,0 +1,613 @@
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.where(:population, :>=, 1000000)
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def where field, operator, value
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          new_query.where ||= default_filter
+          new_query.where.composite_filter.filters << \
+            filter(field, operator, value)
+
+          Query.start new_query, parent_path, client
+        end
+
+        ##
+        # Specifies an "order by" clause on a field.
+        #
+        # @param [String, Symbol] name A field mask to order results with.
+        # @param [String, Symbol] direction The direction to order the results
+        #   by. Values that start with "a" are considered `ascending`. Values
+        #   that start with "d" are considered `descending`. Default is
+        #   `ascending`. Optional.
+        #
+        # @return [Query] New query with `order` called on it.
+        #
+        # @example
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.order(:name)
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        # @example Order by name descending:
+        #   require "google/cloud/firestore"
+        #
+        #   firestore = Google::Cloud::Firestore.new
+        #
+        #   # Get a collection reference
+        #   cities_col = firestore.col "cities"
+        #
+        #   # Create a query
+        #   query = cities_col.order(:name, :desc)
+        #
+        #   query.get do |city|
+        #     puts "#{city.document_id} has #{city[:population]} residents."
+        #   end
+        #
+        def order name, direction = :asc
+          new_query = @query.dup
+          new_query ||= StructuredQuery.new
+
+          new_query.order_by << StructuredQuery::Order.new(
+            field: StructuredQuery::FieldReference.new(field_path: name.to_s),
+            direction: order_direction(direction))
+
+          Query.start new_query, parent_path, client
+        end
+        alias_method :order_by, :order
+
+        ##
+        # Skips to an offset in a query. If the current query already has
+        # specified an offset, this will overwrite it.
+        #
+        # @param [Integer] num The number of results to skip.
+        #
+        # @return [Query] New query with `offset` called on it.
```

Comment:
BTW, thank you for those docs! They are super helpful for my review (and for the users as well).